src/Npgsql/BackendMessages/RowDescriptionMessage.cs

```diff
@@ -350,34 +350,39 @@ internal FieldDescription Clone()
         return field;
     }
 
-    internal void GetInfo(Type? type, ref PgConverterInfo lastConverterInfo)
+    internal void GetInfo(Type? type, ref PgConverterInfo lastConverterInfo, out bool asObject)
     {
         Debug.Assert(lastConverterInfo.IsDefault || (
             ReferenceEquals(_serializerOptions, lastConverterInfo.TypeInfo.Options) &&
             lastConverterInfo.TypeInfo.PgTypeId == _serializerOptions.ToCanonicalTypeId(PostgresType)), "Cache is bleeding over");
 
         if (!lastConverterInfo.IsDefault && lastConverterInfo.TypeToConvert == type)
+        {
+            asObject = lastConverterInfo.IsBoxingConverter;
             return;
+        }
 
         // Have to check for null as it's a sentinel value used by ObjectOrDefaultTypeInfo init itself.
         if (type is not null && ObjectOrDefaultInfo is var odfInfo)
         {
             if (typeof(object) == type)
             {
-                lastConverterInfo = odfInfo with { AsObject = true };
+                lastConverterInfo = odfInfo;
```

Comment:
nit: can be deduped out of the condition

Reply:
Not without perf loss, we'd be doing a copy into lastConverterInfo before we know we hit any of the inner branches that do an early return.

Reply:
You're right, missed that we continue through if both conditions don't match.